tasks/configure.yml

```diff
@@ -158,8 +159,51 @@
     enabled: yes
   when: postgresql_service_enabled
 
-- name: PostgreSQL | Restart PostgreSQL
+- name: PostgreSQL | Ensure PostgreSQL is running
+  service:
+    name: "{{ postgresql_service_name }}"
+    state: started
+
+- name: PostgreSQL | Reload PostgreSQL to implement parameter changes | v9.5 and later
+  service:
+    name: "{{ postgresql_service_name }}"
+    state: reloaded
+  when: (
+          (postgresql_configuration_pt1.changed or
+           postgresql_configuration_pt2.changed or
+           postgresql_configuration_pt3.changed) and
+          postgresql_version | version_compare('9.5','>=')
+        )
+
+- name: PostgreSQL | Check for parameter changes that require a restart | v9.5 and later
+  become: yes
+  become_user: "{{ postgresql_service_user }}"
+  shell: "{{ postgresql_bin_directory}}/psql postgres --username {{postgresql_admin_user}} --tuples-only --command='SELECT count(*) COUNT FROM pg_settings WHERE pending_restart IS true;'"
+  register: postgresql_param_change_restart_required
+  when: postgresql_version | version_compare('9.5','>=')
+  changed_when: postgresql_param_change_restart_required.rc == 0 and
+                postgresql_param_change_restart_required.stdout | replace (' ','') != "0"
+
+- name: PostgreSQL | Force a checkpoint before a restart
```

Comment:
Doing a checkpoint before a restart will ensure the downtime is as short as possible.  The normal (aka "fast") PostgreSQL shutdown process goes something like:

* Shutdown command
* Block all new sessions
* Disconnect and rollback all current sessions
* Flush all dirty blocks to disk
* Shutdown the database "clean"

A checkpoint will force all dirty blocks to be flushed to disk *before* any user sessions are interrupted, so it shortens this step quite significantly for a busy database.